providers/implementations/ciphers/ciphercommon.c

```diff
@@ -314,6 +411,15 @@ int cipher_generic_stream_update(void *vctx, unsigned char *out, size_t *outl,
     }
 
     *outl = inl;
+    /* Remove any TLS padding */
+    if (!ctx->enc && ctx->removetlspad > 0) {
```

Comment:
Is this only supposed to be used by cipher_aes_cbc_hmac_sha1_hw.c & cipher_aes_cbc_hmac_sha256_hw.c.
Perhaps a comment would be good if this is the case.

Reply:
Comment added